catalog/spatial/wfs/spatial-wfs-metacardtype-registry/src/test/java/org/codice/ddf/spatial/ogc/wfs/catalog/metacardtype/registry/impl/WfsMetacardTypeRegistryTest.java

```diff
@@ -94,13 +96,46 @@ public void testLookupMetacardTypeBySimpleName() {
   }
 
   @Test
-  public void testClear() {
+  public void testClearAll() {
     wfsMetacardTypeRegistry.registerMetacardType(
         mockMetacardType, TEST_SOURCE_ID, TEST_FEATURE_SIMPLE_NAME);
-    wfsMetacardTypeRegistry.clear();
+    wfsMetacardTypeRegistry.clearAll();
     verify(mockServiceRegistration, times(1)).unregister();
   }
 
+  @Test
+  public void testClear() {
+    ServiceRegistration mockServiceRegistration1 = mock(ServiceRegistration.class);
+    ServiceRegistration mockServiceRegistration2 = mock(ServiceRegistration.class);
+    ServiceReference mockServiceReference1 = mock(ServiceReference.class);
+    ServiceReference mockServiceReference2 = mock(ServiceReference.class);
+    String featureA = "featureA";
+    String featureB = "featureB";
+
+    when(mockServiceRegistration1.getReference()).thenReturn(mockServiceReference1);
+    when(mockServiceRegistration2.getReference()).thenReturn(mockServiceReference2);
+    when(mockServiceReference1.getProperty(WfsMetacardTypeRegistryImpl.SOURCE_ID))
+        .thenReturn(TEST_SOURCE_ID);
+    when(mockServiceReference2.getProperty(WfsMetacardTypeRegistryImpl.SOURCE_ID))
+        .thenReturn(TEST_SOURCE_ID_2);
+    when(mockServiceReference1.getProperty(WfsMetacardTypeRegistryImpl.FEATURE_NAME))
```

Comment:
:pencil2: The feature name stubs can be removed since they won't be called during this test. These feature names don't match the ones used to register the metacard types anyway.